restapi/src/main/java/io/stargate/web/docsapi/service/JsonConverter.java

```diff
@@ -65,7 +65,7 @@ private JsonNode convertToJsonDoc(
 
     for (Row row : rows) {
       Long rowWriteTime = row.getLong(writeTimeCol.name());
-      String rowLeaf = row.getString("leaf");
+      String rowLeaf = DocumentDB.decodeEscapableChars(row.getString("leaf"));
```

Comment:
Would it not be easier to keep document contents "as is" (unescaped), but support special char escaping when dealing with paths (where `[` and other "forbidden" charts have special meaning)?

Reply:
There are a few cases where I think things might fall apart...I remember thinking about this a while back and realizing that it wouldn't work sometimes to do it the simple way.

If we allow all characters - imagine you have a document A like `{ "a": { "b.c": "value" }}`
that would be stored as p0 = 'a', p1 = 'b.c'

And you would likely want to query it like `{ "a.b.c": {"$eq": "value"}}`

But also, imagine you have a document B `{ "a": { "b": { "c": "value" }}}`
that would be stored as p0 = 'a', p1 = 'b', p2 = 'c'

And you would ALSO want to query it like `{ "a.b.c": {"$eq": "value"}}`

So then there is some ambiguity; however with my change you could theoretically search like
`{ "a.b%2ec": {"$eq": "value"}}` if you want to match document A, and you could do `{ "a.b.c": {"$eq": "value"}}` to match document B.  It's a bit ugly, but removes the ambiguity.

The other situation that is nasty is that if we allow all characters without encoding, someone could save data like `{"[000001]": "a"}` and it would get read out as an array inadvertently - but by encoding the literal value in the database becomes `%5b000001%5d` and so it won't be interpreted as an array (while legitimate array values will have non-encoded `[ ]` in them)

Reply:
These are just two examples, I'm guessing there is a whole lot of nasty cases that could be reached as well...and I agree that this implementation isn't necessarily simple.

To take a step back, the value of doing this is really that people can save keys in their documents that have `.` in them, which is very common since people love to key on emails, e.g. `{ "eric@gmail.com": { "enabled": true }}` or something similar. But I'm second-guessing now whether it's worth the complexity required to accept this?

Reply:
ah I think I see what you're saying about that first case - we could just have url encoding be supported in the filter and then store as-is for that case.

I think the array case is an issue still though

Reply:
> So then there is some ambiguity; however with my change you could theoretically search like
{ "a.b%2ec": {"$eq": "value"}} if you want to match document A, and you could do { "a.b.c": {"$eq": "value"}} to match document B. It's a bit ugly, but removes the ambiguity.

Correct, but I think we do not have to encode stored values for this to work. We only need to parse the path specification from JSON into POJOs correctly, eh?

Reply:
> The other situation that is nasty is that if we allow all characters without encoding, someone could save data like {"[000001]": "a"} and it would get read out as an array inadvertently - but with encoding the literal value in the database becomes %5b000001%5d and so it won't be interpreted as an array (while legitimate array values will have non-encoded [ ] in them)

Good point. But could we not solve this by forbidding `[` and `]` as the first and last property name chars specifically? I'd guess it might be an acceptable restrictions if we could avoid unescaping property names on every read... WDYT?

I do not object, just bringing up points for discusstion :)

Reply:
Yea I'm leaning towards what you just said: `forbidding \[[0-9]+\] (regex) property names` and then allowing data to be written as-is to the database...and then parsing the `where` filter has to deal with encoding of special chars - that would work way better!